### PR TITLE
Rauno devel roslibjs nodejs

### PR DESCRIPTION
**Pull request by rauno99**

### CODE REVIEW

**Comment by kruusamae** [comment]
@Houman-HM please test this on any of the robotonts at your disposal. provide feedback if the README is confusing or you encounter undocumented issues

**Comment by kruusamae** [comment]
@sandra-schumann please test this on the robotont at your disposal. provide feedback if the README is confusing or you encounter undocumented issues. thanks!

**Comment by kruusamae** [comment]
ping @sandra-schumann @Houman-HM !

**Comment by sandra-schumann** [comment]
Webapp works well enough in Chromium, ate up all of my computer's memory (probably) and eventually rendered it unusable in Firefox. Performance is somewhat laggy in both, but camera shows a picture, I could at least once see a depth cloud and the robot can be controlled.

Installation from readme is mostly straightforward, but some users (me) might initially not clone the code into a catkin workspace and realize this should have been done at last step (roslaunch the launch file), which is when they might move the whole thing into a catkin workspace and run catkin build separately (and source devel/setup.bash) to get it to work. The fact that catkin build seems to be part of the setup script (but source'ing setup.bash is not) is somewhat confusing.

**Review by Veix123 — CHANGES_REQUESTED** [review]
I think before merging to main, it would make sense to transition to noetic. depthcloud_encoder [ has been discarded](https://github.com/RobotWebTools/depthcloud_encoder/issues/17) at the end of 2021 and is not available in noetic. Other prerequisites installed ok.  @rauno99 any idea of a replacement for this dependency?
Cloning the repo to workspace still compiles as a workaround.
I also got:
`./ttyd: error while loading shared libraries: libwebsockets.so.8: cannot open shared object file: No such file or directory` 

To sum up:
* webserver comes up nicely
* cmd_vel reaches the driver well
* the ttyd_terminal node shuts down due to the above error.
* wecam stream good
* depthcam stream is present but experienced ~4s delay in the stream
* unable to restart robotont.service due to sudo password being asked message, the sudo workaround has to be added to the readme

**Comment by Veix123** [comment]
I have merged many commits into this PR that made stuff work in noetic, @sudo-Cthulhufhtagn please confirm that it also runs on your robotont so that it can be merged to main.

**Comment by sudo-Cthulhufhtagn** [comment]
The issue with NODE_OPTIONS resolved, all working

**Review by Veix123 — APPROVED** [review]
